File: java/src/main/java/io/hron/ast/IntervalUnit.java
package io.hron.ast;

/** Represents the unit of an interval (minutes or hours). */
public enum IntervalUnit {
  MINUTES("min"),
  HOURS("hours");

  private final String displayName;

  IntervalUnit(String displayName) {
    this.displayName = displayName;
  }

  @Override
  public String toString() {
    return displayName;
  }

  /**
   * Returns the display string based on interval value.
   *
   * @param interval the interval value
   * @return the display string
   */
  public String display(int interval) {
    return switch (this) {
      case MINUTES -> interval == 1 ? "minute" : "min";
      case HOURS -> interval == 1 ? "hour" : "hours";
    };
  }
}

File: csharp/Hron/Ast/IntervalUnit.cs
namespace Hron.Ast;

/// <summary>
/// Represents the unit of an interval (minutes or hours).
/// </summary>
public enum IntervalUnit
{
    Minutes,
    Hours
}

public static class IntervalUnitExtensions
{
    /// <summary>
    /// Returns the display string based on interval value.
    /// </summary>
    public static string Display(this IntervalUnit unit, int interval) => unit switch
    {
        IntervalUnit.Minutes => interval == 1 ? "minute" : "min",
        IntervalUnit.Hours => interval == 1 ? "hour" : "hours",
        _ => throw new ArgumentOutOfRangeException(nameof(unit))
    };
}
